File: MiniDOMAsteriskLib/CallManagers/ManagerLogoutEventArgs.cs
using System;

namespace minidom.CallManagers
{

    /// <summary>
    /// Informazioni sull'accesso ad un callmanager
    /// </summary>
    /// <remarks></remarks>
    public class ManagerLogoutEventArgs : EventArgs
    {
        private string m_UserName;
        private string m_Status;

        public ManagerLogoutEventArgs()
        {
            DMDObject.IncreaseCounter(this);
            m_UserName = "";
        }

        public ManagerLogoutEventArgs(string userName, string status) : this()
        {
            m_UserName = userName;
            m_Status = status;
        }

        ~ManagerLogoutEventArgs()
        {
            DMDObject.DecreaseCounter(this);
        }

        public string UserName
        {
            get
            {
                return m_UserName;
            }
        }

        public string Status
        {
            get
            {
                return m_Status;
            }
        }
    }
}

File: MiniDOMAsteriskLib/CallManagers/ManagerLogoutEventArgs.vb
Imports System.Net.Sockets
Imports System.Threading
Imports System.Net

Namespace CallManagers

    ''' <summary>
    ''' Informazioni sull'accesso ad un callmanager
    ''' </summary>
    ''' <remarks></remarks>
    Public Class ManagerLogoutEventArgs
        Inherits System.EventArgs

        Private m_UserName As String
        Private m_Status As String

        Public Sub New()
            DMDObject.IncreaseCounter(Me)
            Me.m_UserName = ""
        End Sub

        Public Sub New(ByVal userName As String, ByVal status As String)
            Me.New
            Me.m_UserName = userName
            Me.m_Status = status
        End Sub

        Protected Overrides Sub Finalize()
            MyBase.Finalize()
            DMDObject.DecreaseCounter(Me)
        End Sub

        Public ReadOnly Property UserName As String
            Get
                Return Me.m_UserName
            End Get
        End Property

        Public ReadOnly Property Status As String
            Get
                Return Me.m_Status
            End Get
        End Property



    End Class

End Namespace